src/lib.rs

```diff
@@ -4,11 +4,110 @@
 #![feature(thread_local)]
 #![allow(unused_imports)]
 #![feature(hash_extract_if)]
+#![allow(dead_code)]
 
 // interface and safeposix are public because otherwise there isn't a great
 // way to 'use' them for benchmarking.
 pub mod interface;
-// mod lib_fs_utils;
 pub mod safeposix;
 pub mod tests;
 pub mod fdtables;
+
+use crate::safeposix::dispatcher::*;
+
+pub fn lind_lindrustinit(verbosity: i32) {
+    unsafe {
+        lindrustinit(verbosity as isize);
+    }
+}
+
+pub fn lind_lindrustfinalize() {
+    unsafe {
+        lindrustfinalize();
+    }
+}
+
+pub fn lind_rustposix_thread_init(cageid: u64, signalflag: u64) {
+    unsafe {
+        rustposix_thread_init(cageid, signalflag);
+    }
+}
+
+pub fn lind_fork(parent_cageid: u64, child_cageid: u64) -> i32 {
```

Comment:
add comments to explain why those three are special

Reply:
I thought we were removing these?

Reply:
IMO all of this should just be consolidated through the dispatcher, and if not it needs a comment with a good reason.

Reply:
These functions in lib.rs (e.g. `lind_fork`, `lind_exit`, `lind_exec`) are used by multi-process in wasmtime for better readability. Since fork, exit and exec syscall is intercepted and handled internally within wasmtime, so I write a wrapper for these three syscalls. In this way, when I need to call the rawposix fork in clone syscall in wasmtime, I can use `lind_fork` instead of something like `lind_syscall_api(cageid, 68, ..., child_cageid, NOTUSED, NOTUSED, ...)`

Reply:
Don't we use lind_syscall_api everywhere else? I see no reason to do this or why the other functions were moved out of the dispatcher. The lib.rs file should look as it did before.

Reply:
lib.rs is now cleaned up